data_diff/joindiff_tables.py

```diff
@@ -157,7 +157,7 @@ def _diff_tables_root(self, table1: TableSegment, table2: TableSegment, info_tre
             drop_table(db, self.materialize_to_table)
 
         with self._run_in_background(*bg_funcs):
-            if isinstance(db, (Snowflake, BigQuery)):
+            if isinstance(db, (Snowflake, BigQuery, DuckDB)):
```

Comment:
I'm thinking we should have this if/else make a decision based on Database vs. ThreadedDatabase

That's the real decision being made -- not sure why this is sorta hardcoded

Ex:
```python
class MsSQL(ThreadedDatabase):
```

vs.

```python
class Snowflake(Database):

class DuckDB(Database):
```

That's the reason DuckDB was missed here imo


```suggestion
            if not isinstance(db, ThreadedDatabase):
```

Reply:
To clarify: this `if` is deciding whether to joindiff as one (potentially huge) segment/query or multiple parallelizable segments/queries (which the current implementation of duckdb would not really support)

Reply:
Got it, I'll leave that refactor alone for another PR, so that the `ThreadedDatabase` naming convention is consistent across the codebase vs. in this one place.